### ISSUE TITLE
Error in RHEL-07-010491

### ISSUE THREAD

**Issue by danbarr** [ansible-lockdown/RHEL7-STIG#371]
Figured just submitting this as an issue might be quicker than a PR given the workaround you have to do for the CI permissions...

RHEL-07-010491, the fix for UEFI systems (second task in the block - line 225 in the current develop version of fix-cat1.yml) inserts a ^ at the beginning of the line; it should be there in the regexp: line but removed from the inserted text.

https://github.com/ansible-lockdown/RHEL7-STIG/blob/eb8d5bcf1b756b82fc272dee6f6a27245e49c5e0/tasks/fix-cat1.yml#L225

Current:
```
    - regexp: ^GRUB2_PASSWORD=password_pbkdf2 {{ rhel7stig_boot_superuser }}
      line: ^GRUB2_PASSWORD=password_pbkdf2 {{ rhel7stig_boot_superuser }} {{ rhel7stig_bootloader_password_hash }}
```

Fixed:
```
    - regexp: ^GRUB2_PASSWORD=password_pbkdf2 {{ rhel7stig_boot_superuser }}
      line: GRUB2_PASSWORD=password_pbkdf2 {{ rhel7stig_boot_superuser }} {{ rhel7stig_bootloader_password_hash }}
```

**Comment by danbarr** [comment]
Another suggestion for the same task, to make it compatible with CentOS:

Replace:
https://github.com/ansible-lockdown/RHEL7-STIG/blob/eb8d5bcf1b756b82fc272dee6f6a27245e49c5e0/tasks/fix-cat1.yml#L218

With:
`      path: /boot/efi/EFI/{{ ansible_distribution | lower }}/user.cfg`

**Comment by danbarr** [comment]
Oof, discovered another issue with this one which affects the UEFI version.

It's looking for and setting "GRUB2_PASSWORD=password_pbkdf2 root <hash>"

But this doesn't work; the syntax is just "GRUB2_PASSWORD=<hash>", like "GRUB2_PASSWORD=grub.pbkdf2.sha512.10000.HASHSTRING". I verified this using grub2-setpassword.

The BIOS version has it right.

Replace:
```
    with_items:
    - regexp: ^GRUB2_PASSWORD=password_pbkdf2 {{ rhel7stig_boot_superuser }}
      line: GRUB2_PASSWORD=password_pbkdf2 {{ rhel7stig_boot_superuser }} {{ rhel7stig_bootloader_password_hash }}
```

With:
```
    with_items:
    - regexp: ^GRUB2_PASSWORD={{ rhel7stig_bootloader_password_hash }}
      line: GRUB2_PASSWORD={{ rhel7stig_bootloader_password_hash }}
```

I'll just submit a PR for all 3 of these :)

**Comment by danbarr** [comment]
I see this was mentioned in #374, but the fix is not quite right:

https://github.com/ansible-lockdown/RHEL7-STIG/blob/a1c98bba089256bd86a7c14537e01a9bc7c29b67/tasks/fix-cat1.yml#L224-L225

Should be: 
```
            - regexp: ^GRUB2_PASSWORD={{ rhel7stig_bootloader_password_hash }}
              line: GRUB2_PASSWORD={{ rhel7stig_bootloader_password_hash }}
```

**Comment by georgenalen** [comment]
@danbarr 
 Thanks for this catch, I'm working on getting this fixed moved through. 

George

**Comment by georgenalen** [comment]
Fixed in release 1.2.1.
